idp/client.ts

```diff
@@ -0,0 +1,45 @@
+import { APIGatewayProxyEvent, APIGatewayProxyResult, Context } from "aws-lambda";
+import axios from "axios";
+import { Logger } from "@aws-lambda-powertools/logger";
+import { SecretsManager } from "@aws-sdk/client-secrets-manager";
+const logger = new Logger({ serviceName: "TestIdp" });
+
+const secrets = new SecretsManager({ useFipsEndpoint: true });
+
+const IDP_CLIENT_ID = process.env.IDP_CLIENT_ID!;
+const IDP_CLIENT_SECRET_PATH = process.env.IDP_CLIENT_SECRET_PATH!;
+const IDP_DOMAIN = process.env.IDP_DOMAIN!;
+
+export async function handler(event: APIGatewayProxyEvent, context: Context): Promise<APIGatewayProxyResult> {
+  logger.addContext(context);
+  logger.info("Retriving client secret from Secrets Manager", { secretId: IDP_CLIENT_SECRET_PATH });
+  const clientSecret = (await secrets.getSecretValue({ SecretId: IDP_CLIENT_SECRET_PATH })).SecretString!;
+  logger.info("Retrieving a token from IdP using Client ID", { clientId: IDP_CLIENT_ID });
+  const form = new FormData();
+  form.set("grant_type", "client_credentials");
+  form.set("client_id", IDP_CLIENT_ID);
+  const response = await axios.post(`https://${IDP_DOMAIN}/oauth2/token`, form, {
+    headers: {
```

Comment:
Should there be a time out for the `await`?

Reply:
There should be a timeout, though to nitpick a bit, it's not the `await` but for the HTTP request as a whole.